// @codux-boards/react-semantic/boards/Grid/stretched.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Grid, Segment } from 'semantic-ui-react';

// // board cover
import cover from './stretched.png';

export default createBoard({
    name: 'Stretched',
    Board: () => (
        <div style={{ width: '450px' }}>
            <Grid columns={3} divided>
                <Grid.Row stretched>
                    <Grid.Column>
                        <Segment>1</Segment>
                    </Grid.Column>
                    <Grid.Column>
                        <Segment>1</Segment>
                        <Segment>2</Segment>
                    </Grid.Column>
                    <Grid.Column>
                        <Segment>1</Segment>
                        <Segment>2</Segment>
                        <Segment>3</Segment>
                    </Grid.Column>
                </Grid.Row>
            </Grid>
        </div>
    ),
    tags: ['accordion', 'icon'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
