// @codux-boards/react-semantic/boards/Card/card.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Card, Icon, Image } from 'semantic-ui-react';

// board cover
import cover from './card.png';

export default createBoard({
    name: 'Card',
    Board: () => (
        <Card>
            <Image
                src="https://react.semantic-ui.com/images/avatar/large/matthew.png"
                wrapped
                ui={false}
            />
            <Card.Content>
                <Card.Header>Matthew</Card.Header>
                <Card.Meta>
                    <span className="date">Joined in 2015</span>
                </Card.Meta>
                <Card.Description>Matthew is a musician living in Nashville.</Card.Description>
            </Card.Content>
            <Card.Content extra>
                <a>
                    <Icon name="user" />
                    22 Friends
                </a>
            </Card.Content>
        </Card>
    ),
    tags: ['card', 'icon', 'image'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
