// @codux-boards/react-semantic/boards/Grid/basic.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Grid, Image } from 'semantic-ui-react';

// board cover
import cover from './basic.png';

export default createBoard({
    name: 'Basic',
    Board: () => (
        <div style={{ width: '450px' }}>
            <Grid centered padded columns={2}>
                <Grid.Row>
                    <Grid.Column>
                        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" />
                    </Grid.Column>
                    <Grid.Column>
                        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" />
                    </Grid.Column>
                </Grid.Row>
                <Grid.Row>
                    <Grid.Column>
                        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" />
                    </Grid.Column>
                    <Grid.Column>
                        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" />
                    </Grid.Column>
                </Grid.Row>
            </Grid>
        </div>
    ),
    tags: ['image', 'grid'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
