// @codux-boards/react-semantic/boards/Forms/form-subcomponents.board.tsx
import { createBoard } from '@wixc3/react-board';
import { useState } from 'react';
import { Form } from 'semantic-ui-react';

// board cover
import cover from './form-subcomponents.png';

const options = [
    { key: 'm', text: 'Male', value: 'male' },
    { key: 'f', text: 'Female', value: 'female' },
    { key: 'o', text: 'Other', value: 'other' },
];

export default createBoard({
    name: 'Form Subcomponents',
    Board: () => {
        const [selectedRadio, setSelectedRadio] = useState<string>();

        const handleChange = (value: string) => setSelectedRadio(value);

        return (
            <Form>
                <Form.Group widths="equal">
                    <Form.Input fluid label="First name" placeholder="First name" />
                    <Form.Input fluid label="Last name" placeholder="Last name" />
                    <Form.Select fluid label="Gender" options={options} placeholder="Gender" />
                </Form.Group>
                <Form.Group inline>
                    <label>Size</label>
                    <Form.Radio
                        label="Small"
                        value="sm"
                        checked={selectedRadio === 'sm'}
                        onChange={() => handleChange('sm')}
                    />
                    <Form.Radio
                        label="Medium"
                        value="md"
                        checked={selectedRadio === 'md'}
                        onChange={() => handleChange('md')}
                    />
                    <Form.Radio
                        label="Large"
                        value="lg"
                        checked={selectedRadio === 'lg'}
                        onChange={() => handleChange('lg')}
                    />
                </Form.Group>
                <Form.TextArea label="About" placeholder="Tell us more about you..." />
                <Form.Checkbox label="I agree to the Terms and Conditions" />
                <Form.Button>Submit</Form.Button>
            </Form>
        );
    },
    tags: ['Button', 'Checkbox', 'Form', 'Subcomponents'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
