// @codux-boards/react-semantic/boards/Step/steps.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Icon, Step } from 'semantic-ui-react';

// board cover
import cover from './steps.png';

export default createBoard({
    name: 'Steps',
    Board: () => (
        <Step.Group>
            <Step>
                <Icon name="truck" />
                <Step.Content>
                    <Step.Title>Shipping</Step.Title>
                    <Step.Description>Choose your shipping options</Step.Description>
                </Step.Content>
            </Step>

            <Step active>
                <Icon name="payment" />
                <Step.Content>
                    <Step.Title>Billing</Step.Title>
                    <Step.Description>Enter billing information</Step.Description>
                </Step.Content>
            </Step>

            <Step disabled>
                <Icon name="info" />
                <Step.Content>
                    <Step.Title>Confirm Order</Step.Title>
                </Step.Content>
            </Step>
        </Step.Group>
    ),
    tags: ['Icon', 'Steps'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
