// @codux-boards/react-semantic/boards/Item/item.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Image, Item } from 'semantic-ui-react';

// board cover
import cover from './item.png';

export default createBoard({
    name: 'Item',
    Board: () => (
        <Item.Group>
            <Item>
                <Item.Image
                    size="tiny"
                    src="https://react.semantic-ui.com/images/wireframe/image.png"
                />

                <Item.Content>
                    <Item.Header>Header</Item.Header>
                    <Item.Meta>Description</Item.Meta>
                    <Item.Description>
                        <Image src="https://react.semantic-ui.com/images/wireframe/short-paragraph.png" />
                    </Item.Description>
                    <Item.Extra>Additional Details</Item.Extra>
                </Item.Content>
            </Item>

            <Item>
                <Item.Image
                    size="tiny"
                    src="https://react.semantic-ui.com/images/wireframe/image.png"
                />

                <Item.Content>
                    <Item.Header>Header</Item.Header>
                    <Item.Meta>Description</Item.Meta>
                    <Item.Description>
                        <Image src="https://react.semantic-ui.com/images/wireframe/short-paragraph.png" />
                    </Item.Description>
                    <Item.Extra>Additional Details</Item.Extra>
                </Item.Content>
            </Item>
        </Item.Group>
    ),
    tags: ['Image', 'Item'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
        canvasWidth: 500,
    },
});
