// @codux-boards/react-semantic/boards/Dropdown/file.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Dropdown } from 'semantic-ui-react';

// board cover
import cover from './file.png';

export default createBoard({
    name: 'File',
    Board: () => (
        <Dropdown text="File">
            <Dropdown.Menu>
                <Dropdown.Item text="New" />
                <Dropdown.Item text="Open..." description="ctrl + o" />
                <Dropdown.Item text="Save as..." description="ctrl + s" />
                <Dropdown.Item text="Rename" description="ctrl + r" />
                <Dropdown.Item text="Make a copy" />
                <Dropdown.Item icon="folder" text="Move to folder" />
                <Dropdown.Item icon="trash" text="Move to trash" />
                <Dropdown.Divider />
                <Dropdown.Item text="Download As..." />
                <Dropdown.Item text="Publish To Web" />
                <Dropdown.Item text="E-mail Collaborators" />
            </Dropdown.Menu>
        </Dropdown>
    ),
    tags: ['Dropdown', 'File'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 30, left: 30 },
        windowBackgroundColor: '#fcfcff',
    },
});
