// @codux-boards/react-semantic/boards/Accordion/styled.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Accordion } from 'semantic-ui-react';

// board cover
import cover from './styled.png';

export default createBoard({
    name: 'Styled',
    Board: () => (
        <Accordion
            defaultActiveIndex={0}
            panels={[
                {
                    key: 'panel-1',
                    title: 'What is a dog?',
                    content: (
                        <Accordion.Content>
                            <p>
                                A dog is a type of domesticated animal. Known for its loyalty and
                                faithfulness, it can be found as a welcome guest in many households
                                across the world.
                            </p>
                        </Accordion.Content>
                    ),
                },
                {
                    key: 'panel-2',
                    title: 'What kinds of dogs are there?',
                    content: (
                        <Accordion.Content>
                            <p>
                                There are many breeds of dogs. Each breed varies in size and
                                temperament. Owners often select a breed of dog that they find to be
                                compatible with their own lifestyle and desires from a companion.
                            </p>
                        </Accordion.Content>
                    ),
                },
                {
                    key: 'panel-3',
                    title: 'How do you acquire a dog?',
                    content: (
                        <Accordion.Content>
                            <p>
                                Three common ways for a prospective owner to acquire a dog is from
                                pet shops, private owners, or shelters.
                            </p>
                            <p>
                                A pet shop may be the most convenient way to buy a dog. Buying a dog
                                from a private owner allows you to assess the pedigree and
                                upbringing of your dog before choosing to take it home. Lastly,
                                finding your dog from a shelter, helps give a good home to a dog who
                                may not find one so readily.
                            </p>
                        </Accordion.Content>
                    ),
                },
            ]}
            styled
        />
    ),
    tags: ['accordion', 'icon'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
