// @codux-boards/react-semantic/boards/Divider/vertical-divider.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Button, Divider, Form, Grid, Segment } from 'semantic-ui-react';

// board cover
import cover from './vertical-divider.png';

export default createBoard({
    name: 'Vertical Divider',
    Board: () => (
        <Segment placeholder>
            <Grid columns={2} relaxed="very" stackable>
                <Grid.Column>
                    <Form>
                        <Form.Input
                            icon="user"
                            iconPosition="left"
                            label="Username"
                            placeholder="Username"
                        />
                        <Form.Input
                            icon="lock"
                            iconPosition="left"
                            label="Password"
                            type="password"
                        />

                        <Button content="Login" primary />
                    </Form>
                </Grid.Column>

                <Grid.Column verticalAlign="middle">
                    <Button content="Sign up" icon="signup" size="big" />
                </Grid.Column>
            </Grid>

            <Divider vertical>Or</Divider>
        </Segment>
    ),
    tags: ['Button', 'Divider', 'Form', 'Grid', 'Segment'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
    },
});
