// @codux-boards/react-semantic/boards/Sidebar/sidebar.board.tsx
import { useState } from 'react';
import { createBoard } from '@wixc3/react-board';
import { Button, Header, Image, Menu, Segment, Sidebar } from 'semantic-ui-react';

// board cover
import cover from './sidebar.png';

export default createBoard({
    name: 'Sidebar',
    Board: () => {
        const [visible, setVisible] = useState(true);
        return (
            <div>
                <Button icon="sidebar" onClick={() => setVisible(true)} />
                <Sidebar.Pushable>
                    <Sidebar
                        as={Menu}
                        animation="overlay"
                        icon="labeled"
                        inverted
                        onHide={() => setVisible(false)}
                        vertical
                        visible={visible}
                        width="thin"
                    >
                        <Menu.Item as="a" icon="home" content="Home" />
                        <Menu.Item as="a" icon="gamepad" content="Games" />
                        <Menu.Item as="a" icon="camera" content="Channels" />
                    </Sidebar>
                    <Sidebar.Pusher>
                        <Segment basic>
                            <Header as="h3">Application Content</Header>
                            <Image src="https://react.semantic-ui.com/images/wireframe/paragraph.png" />
                        </Segment>
                    </Sidebar.Pusher>
                </Sidebar.Pushable>
            </div>
        );
    },
    tags: ['Checkbox', 'Grid', 'Header', 'Icon', 'Image', 'Menu', 'Segment', 'Sidebar'],
    cover,
    environmentProps: {
        canvasPadding: { top: 10, bottom: 10, right: 10, left: 10 },
        windowBackgroundColor: '#fcfcff',
        canvasWidth: 500,
    },
});
